charts/example-configurations/minikube-values-gitlab.yaml

```diff
@@ -106,7 +106,8 @@ graph:
       renku:
         # At the moment the url matches 'gitlab.url' but it should be the Renku UI url
         # once https://github.com/SwissDataScienceCenter/renku-python/issues/554 issue is played
-        url: https://gitlab.com
+        url: https://dev.renku.ch
```

Comment:
Yes, you're right. However, if we leave `https://gitlab.com` all the knowledge-graph queries will be returning no results. By the way, maybe you know what's the default value `renku-cli` uses for the `isPartOf` when generating the log?

Reply:
Yeah, `url: https://gitlab.com` is definitely wrong too. I think it should be what I've suggested below.

Reply:
Sounds a good solution although not entirely sure if renku-cli picks minikube ip when generating the triples. If not, it should anyway and sooner or later it will ;)

Reply:
you mean to set the node ids?

Reply:
Yes, for the `isPartOf` pointing to a project.

Reply:
ah right - it only sets it if the remote is known - otherwise we currently don't have any good way of setting it and it defaults to `file:///...`

Reply:
Right, so that's a problem as knowledge-graph wouldn't be able to find any triples for a project. Couldn't we use GitLab IP for that? Or if we introduce a resource like knowledge-graph/projects/:id we'd be able to use it on all the environments.

Reply:
why can't it find them? that's just the prefix

Reply:
Yes, but I don't think the query should be matching on part of the resource as it will slow it down dramatically. Why do you think using `http://minikube ip` for the host wouldn't be a better idea?

Reply:
sorry I dropped this discussing - I still don't understand why it's necessary to know the URL prefix to find nodes or to match on a partial prefix?

Reply:
It's just from the query performance point of view. As you know the `isPartOf` triples are URL resources which, for instance on the DEV, looks like `<https://dev.renku.ch/namespace/project>`. The query can try to match triples of a particular project by doing partial matches on `namespace/project` but I imagine it will be much faster if it can simply compare the full URLs. So if on a minikube, the mentioned resources are `<http://minkube IP/namespace/project>`, the `renku.url` can have value of `http://minkube IP` and matches on full URL could be still possible. Does this make sense?

Reply:
but you could do something like

```
SELECT ?project

WHERE {
  ?project schema:name <name>
  ?project schema:creator <creator>
}
```

Or some other descriptive property and the query will get the correct project node for you without knowing the URI prefix. I think in general you want to discover the prefix not hard-code it in.

Reply:
Yep, that would work. I'd be more than happy to go this way and once we have projects described like that, I can simply get rid of the property completely. Should we have a tiny epic for such a change?

Reply:
are there currently no creator/name triples?

Reply:
I don't think so. As far I can see we have just these:
```
  <rdf:Description rdf:about="https://dev.renku.ch/virginiafriedrich/datasets-test">
    <rdf:type rdf:resource="http://www.w3.org/ns/prov#Location"/>
    <rdf:type rdf:resource="http://xmlns.com/foaf/0.1/Project"/>
  </rdf:Description>
```

Reply:
I see - then yes, please make an epic to update this metadata

Reply:
In any case, the default should be the remote base URL, which can be set per deployment but in general should be the same as `global.renku.url` afaik

Reply:
I cannot find `global.renku.url` being defined or set during `helm upgrade --install ...` but `global.renku.domain`. Is that what you mean or there's some hidden `global.renku.url` defined somewhere?

Reply:
Sorry, that's the one I meant

Reply:
Right, so then I think we should be fine as the README after the change suggests:
```--set global.renku.domain=$(minikube ip)
...
--set graph.knowledgeGraph.services.renku.url=http://$(minikube ip)```

Reply:
Should we then leave for the time being and we'll change it once we introduce more metadata to the project triples?

Reply:
yep sounds good

Reply:
I'd just remove `url` and `resources-url` from this values file as the values have to be set through the command line anyway.

Reply:
:+1 Done